### PR TITLE
Closes #2: As a developer, I want my commits to be checked by a linter so that code standards are maintained

### LINKED ISSUE #2: As a developer, I want my commits to be checked by a linter so that code standards are maintained.
**Context**

The wp-rocket-e2e project has a great potential to speed up our development processes: as suggested by @CrochetFeve0251 in #1, every developer could leverage it to automate testing and validation of acceptance criteria. This means that many people will perform small updates regularly on this repository. However, it is currently not up to standards, making it difficult to maintain properly overtime.

Several steps must be taken to make this codebase "production ready".

**Expected behavior**

When a commit is pushed to the repository, a linter must be applied and results must be reported.
The repository must be ready-to-use with a linter locally so that developers can perform checks themselves.

All automatically fixable errors must be fixed. Exceptionally, it is not needed that the linter returns no error for this issue to be closed: several refactoring must occur shortly, so let's not waste time fixing code that will disappear.

**Proposed solution**

We could use eslint:
- It is a well known linter for JS/TS,
- It is easy to set up locally,
- There are github actions available to quickly set up the CI.

### PR DESCRIPTION
**Pull request by jeawhanlee**
Added Linter for E2E Tests

Closes #2

### CODE REVIEW

**Comment by CrochetFeve0251** [comment]
Hey @jeawhanlee,
I saw the PR so I checked a little the code inside it even if I am not sure it is finished work.
While running the linter the library `eslint-plugin-es5` was missing into dev dependencies if you can add it.

Then the second thing I would add is a way to quick fix problems which allows it with the script `"lint:fix": "eslint  . --ext .ts --fix"`

Otherwise look like the linter is working well, congrats!

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "WP Rocket E2E Tests with Playwright",
   "main": "playwright.config.ts",
   "scripts": {
-    "test:e2e": "playwright test --config playwright.config.ts"
+    "test:e2e": "playwright test --config playwright.config.ts",
+    "lint": "eslint . --ext .ts"
```
Comment (CrochetFeve0251) [inline package.json]:
```suggestion
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint  . --ext .ts --fix"
```

File: package.json [hunk 2]
```diff
@@ -13,7 +14,11 @@
   "author": "WP Media",
   "license": "ISC",
   "devDependencies": {
-    "@playwright/test": "^1.32.3"
+    "@playwright/test": "^1.32.3",
+    "@typescript-eslint/eslint-plugin": "^6.1.0",
+    "@typescript-eslint/parser": "^6.1.0",
+    "eslint": "^8.45.0",
```
Comment (CrochetFeve0251) [inline package.json:21]:
```suggestion
    "eslint": "^8.45.0",
    "eslint-plugin-es5": "^1.5.0",
```

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

**Comment by jeawhanlee** [comment]
> Hey @jeawhanlee, I saw the PR so I checked a little the code inside it even if I am not sure it is finished work. While running the linter the library `eslint-plugin-es5` was missing into dev dependencies if you can add it.
> 
> Then the second thing I would add is a way to quick fix problems which allows it with the script `"lint:fix": "eslint . --ext .ts --fix"`
> 
> Otherwise look like the linter is working well, congrats!

Thanks for this @CrochetFeve0251 for compatibility issue with older versions of node, we should add this dependency.

**Review by CrochetFeve0251 — APPROVED** [review]
If you could solve the linter errors that would be great

**Comment by jeawhanlee** [comment]
> If you could solve the linter errors that would be great

@CrochetFeve0251 We will still do a refactor [here](https://github.com/wp-media/wp-rocket-e2e/issues/3) so it would make sense not to resolve linter errors of code that we might not keep.

**Comment by CrochetFeve0251** [comment]
> > If you could solve the linter errors that would be great
> 
> @CrochetFeve0251 We will still do a refactor [here](https://github.com/wp-media/wp-rocket-e2e/issues/3) so it would make sense not to resolve linter errors of code that we might not keep.

@jeawhanlee you are right 👍

**Pull request merged**